pkg/kubelet/status_syncer.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/golang/glog"
+)
+
+type podSyncStatus struct {
+	api.PodStatus
+	synced bool
+}
+
+type statusSyncer struct {
+	kubeClient client.Interface
+	// Map from pod full name to sync status of the corresponding pod.
+	podStatusesLock sync.RWMutex
+	podStatuses     map[string]podSyncStatus
+
+	// TODO(filipg): Remove, unused
+	podStatusUpdateFrequency time.Duration
+}
+
+func newStatusSyncer(kubeClient client.Interface) *statusSyncer {
+	return &statusSyncer{
+		kubeClient:  kubeClient,
+		podStatuses: make(map[string]podSyncStatus),
+	}
+}
+
+func (s *statusSyncer) Run() {
+	// We can run Sync() often because it will only update statuses that have
+	// changed and is cheap if there are none.
+	go util.Forever(s.Sync, 5*time.Second)
+}
+
+func (s *statusSyncer) GetPodStatus(podFullName string) (api.PodStatus, bool) {
+	s.podStatusesLock.RLock()
+	defer s.podStatusesLock.RUnlock()
+	status, ok := s.podStatuses[podFullName]
+	return status.PodStatus, ok
+}
+
+func (s *statusSyncer) SetPodStatus(podFullName string, status api.PodStatus) {
```

Comment:
I'm not sure I understand - are you suggesting to have a go routine that listens on a channel and wait for status updates, instead of having periodic Sync method()? WDYT about ignoring quick status updates for now to simplify logic?